### PR TITLE
Close sidebar on click outside 

### PR DESCRIPTION
**Pull request by alter-eggo**
**Overview**
This pr adds closing sidebar on click outside + small refactoring of sidebar code

### CODE REVIEW

**Comment by Tulun** [comment]
Overall looks ok but I'd like to see this after @james-a-morris comments are addressed

File: src/components/Sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -66,106 +77,27 @@ const Sidebar: FC<Props> = ({ openSidebar, setOpenSidebar }) => {
...
-              href="https://forum.across.to/"
-              target="_blank"
-              rel="noreferrer"
-              onClick={() => onClickLink()}
-            >
-              Discourse
-            </a>
-          </StyledMenuItem>
+          {appMenu.map((item) => (
+            <StyledMenuItem selected={location.pathname === item.pathName}>
```
[the remark this answers is not in the document]
Comment (alter-eggo) [reply src/components/Sidebar/Sidebar.tsx]: sure, done

File: src/components/Sidebar/useSidebar.ts [hunk 2]
```diff
@@ -26,6 +26,59 @@ export default function useSidebar(openSidebar: boolean) {
       }, 250);
     }
   }, [openSidebar, prevOpenSidebar]);
+
+  const appMenu = [
```
[the remark this answers is not in the document]
Comment (alter-eggo) [reply src/components/Sidebar/useSidebar.ts]: nice catch, fixed

**Comment by alter-eggo** [comment]
@james-a-morris @Tulun fixed comments

**Comment by mrice32** [comment]
@Tulun are you good for me to merge this?

**Review by Tulun — APPROVED** [review]

**Comment by mrice32** [comment]
Thanks @alter-eggo!!

**Pull request merged**